### PR TITLE
go 1.14

### PR DESCRIPTION
**Pull request by techknowlogick**
Note: This also increases the minimum supported version of go to 1.12

### CODE REVIEW

**Review by sapk — DISMISSED** [review]
Please update the go.mod version to use latest go std lib.

**Comment by silverwind** [comment]
> This also increases the minimum supported version of go to 1.12

Need to update Makefile, README.md and docs with `1.11` -> `1.12`.

**Comment by techknowlogick** [comment]
Failing due to error described in #10552

**Review by silverwind — APPROVED** [review]

**Comment by techknowlogick** [comment]
Current template that need resolution: 
```
template: repo/issue/list:271:46: executing "repo/issue/list" at <(len .PullRequest.ConflictedFiles) gt 0>: can't give argument to non-function len .PullRequest.ConflictedFiles

```

**Comment by techknowlogick** [comment]
Ok, now that template issues have been resolved, it seems that on ARM there are failures when `TestAPIUpdateFile` is run. I suspect it is related to indexing of code.

**Comment by 6543** [comment]
#10597  got merged can you update/use goversion var ;

**Comment by jolheiser** [comment]
Resolved conflicts. :+1:

**Comment by lafriks** [comment]
looks like some kind of deadlock in queues on arm

File: Makefile [hunk 1]
```diff
@@ -479,7 +479,7 @@ release-windows: | $(DIST_DIRS)
 	@hash xgo > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
 		$(GO) get -u src.techknowlogick.com/xgo; \
 	fi
-	xgo -dest $(DIST)/binaries -tags 'netgo osusergo $(TAGS)' -ldflags '-linkmode external -extldflags "-static" $(LDFLAGS)' -targets 'windows/*' -out gitea-$(VERSION) .
+	xgo -go go-1.14 -dest $(DIST)/binaries -tags 'netgo osusergo $(TAGS)' -ldflags '-linkmode external -extldflags "-static" $(LDFLAGS)' -targets 'windows/*' -out gitea-$(VERSION) .
```
Comment (6543) [inline Makefile]: Can you move the version into a var?

Comment (silverwind) [reply Makefile]: Maybe add `GO_VERSION`, `MIN_GO_VERSION`, `NODE_VERSION`, `MIN_NODE_VERSION` (third one unused in Makefile context).

**Review by sapk — COMMENTED** [review]
Removing my blocking review since this part has been fixed.

**Comment by guillep2k** [comment]
<https://golang.org/doc/go1.14>

> A consequence of the implementation of preemption is that on Unix systems, including Linux and macOS systems, programs built with Go 1.14 will receive more signals than programs built with earlier releases. This means that programs that use packages like syscall or golang.org/x/sys/unix will see more slow system calls fail with EINTR errors. Those programs will have to handle those errors in some way, most likely looping to try the system call again. For more information about this see man 7 signal for Linux systems or similar documentation for other systems.

Should we be concerned about this? How many of our vendored packages use system calls?

**Review by 6543 — APPROVED** [review]

**Comment by silverwind** [comment]
@guillep2k 

``` console
$ grep '"syscall"' vendor -r | wc -l
161
$ grep '"golang.org/x/sys/unix"' vendor -r | wc -l
11
```

**Comment by techknowlogick** [comment]
ping LG-TM

**Pull request merged**